BossMod/Autorotation/RotationModule.cs

```diff
@@ -123,13 +123,22 @@ public bool TraitUnlocked(uint id)
         return status != null ? (StatusDuration(status.Value.ExpireAt), status.Value.Extra & 0xFF) : (0, 0);
     }
     protected (float Left, int Stacks) StatusDetails<SID>(Actor? actor, SID sid, ulong sourceID, float pendingDuration = 1000) where SID : Enum => StatusDetails(actor, (uint)(object)sid, sourceID, pendingDuration);
-
     protected (float Left, int Stacks) SelfStatusDetails(uint sid, float pendingDuration = 1000) => StatusDetails(Player, sid, Player.InstanceID, pendingDuration);
     protected (float Left, int Stacks) SelfStatusDetails<SID>(SID sid, float pendingDuration = 1000) where SID : Enum => StatusDetails(Player, sid, Player.InstanceID, pendingDuration);
 
+    // checks if Status effect is on self
+    // for buffs/effects that have no duration (i.e. 'Eukrasia')
+    protected bool SelfStatusCheck<SID>(SID sid) where SID : Enum => Player.FindStatus((uint)(object)sid, Player.InstanceID) != null;
```

Comment:
So, are these things needed because eg for eukrasia 'left' is always 0, making it impossible to distinguish missing vs non-missing buff?

If so, what is the benefit of this helper vs just calling Player.FindStatus? The reason why StatusDetails etc exist (I don't actually like them here btw, ideally they would be on actor) is because they also check pending statuses (this is needed eg for dots and hots, to avoid casting same thing twice). These helpers do not - and I don't really know whether you intended for them to do so - but the point is that they are called similarly, but behave slightly differently.